### PR TITLE
Fix an Intellij debug issue [databricks]

### PR DESCRIPTION
**Pull request by wbo4958**
Intellij complains below error when debugging

```
imported `TypeSigUtil` is permanently hidden by definition of trait TypeSigUtil in package rapids
import com.nvidia.spark.rapids.shims.v2.TypeSigUtil
```

This PR just renames TypeSigUtil to TypeSigUtilTrait to avoid this.

Signed-off-by: Bobby Wang <wbo4958@gmail.com>

### CODE REVIEW

**Comment by wbo4958** [comment]
build

**Comment by wbo4958** [comment]
build

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala [hunk 1]
```diff
@@ -25,8 +25,8 @@ import com.nvidia.spark.rapids.shims.v2.TypeSigUtil
 import org.apache.spark.sql.catalyst.expressions.{Attribute, Expression, UnaryExpression, WindowSpecDefinition}
 import org.apache.spark.sql.types._
 
-/** TypeSigUtil for different spark versions */
-trait TypeSigUtil {
+/** TypeSigUtilBase for different spark versions */
```
Comment (firestarman) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala:28]:
```suggestion
/** Base TypeSigUtil for different spark versions */
```

File: sql-plugin/src/main/301until320-all/scala/com/nvidia/spark/rapids/shims/v2/TypeSigUtil.scala [hunk 2]
```diff
@@ -16,12 +16,12 @@
...
+import com.nvidia.spark.rapids.{TypeEnum, TypeSig, TypeSigUtilBase}
 
 import org.apache.spark.sql.types.DataType
 
 /** TypeSig Support for [3.0.1, 3.2.0) */
-object TypeSigUtil extends com.nvidia.spark.rapids.TypeSigUtil {
+object TypeSigUtil extends TypeSigUtilBase {
 
   /**
    * Check if this type of Spark-specific is supported by the plugin or not.
```
Comment (jlowe) [inline sql-plugin/src/main/301until320-all/scala/com/nvidia/spark/rapids/shims/v2/TypeSigUtil.scala:27]: Not caused by this PR, but while reviewing I noticed there's `allowDecimal` scaladoc params that are incorrect in this file.

Comment (wbo4958) [reply sql-plugin/src/main/301until320-all/scala/com/nvidia/spark/rapids/shims/v2/TypeSigUtil.scala:27]: But this PR can help me to debug without any error.

Comment (wbo4958) [reply sql-plugin/src/main/301until320-all/scala/com/nvidia/spark/rapids/shims/v2/TypeSigUtil.scala:27]: will check this para

Comment (wbo4958) [reply sql-plugin/src/main/301until320-all/scala/com/nvidia/spark/rapids/shims/v2/TypeSigUtil.scala:27]: Done

**Comment by wbo4958** [comment]
build

**Comment by wbo4958** [comment]
build

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala (hunk 1 above)
Comment (wbo4958) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala]: Done

**Review by GaryShen2008 — APPROVED** [review]

**Pull request merged**